File: web/src/app/company/Employee.ts
import { Personality } from "../shared/Personality";
import { Skill } from "../shared/skill";
import { TeamShort } from "./Team";

export class Employee {
    public constructor(
        public id: number,
        public name: string,
        public title: string,
        public personality: Personality,
        public skills: Skill[] = [],
        public evaluations: EmployeeEvaluation[] = [],
        public teams: TeamShort[] = [],
    ) {}
}

export class EmployeeEvaluation {
  public constructor(
    public id: number,
    public date: string,
    public result: number,
  ) {}
}

export class EmployeeResponse {
    success: boolean;
    data: Employee;

    public constructor(
        success: boolean,
        data: Employee
    ) {
        this.success = success;
        this.data = data
    }
}

export class EmployeesListResponse {
    success: boolean;
    data: Employee[];

    public constructor(
        success: boolean,
        data: Employee[]
    ) {
        this.success = success;
        this.data = data
    }
}

export class EmployeeCreationRequest {
    name: string;
    title: string;
    personality_id: number;
    skills: number[];

    public constructor(
        name: string,
        title: string,
        personality: Personality,
        skills: number[]
    ) {
        this.name = name;
        this.title = title;
        this.personality_id = personality ? personality.id : 0;
        this.skills = skills;
    }
}

File: mobile/app/src/main/java/com/example/abc_jobs_alpaca/model/models/Employee.kt
package com.example.abc_jobs_alpaca.model.models

import com.google.gson.Gson
import com.google.gson.annotations.SerializedName
import org.json.JSONObject
import java.util.Date

data class Employee(
    val id: Int,
    @SerializedName("id_persona") val idPersona: Int,
    val name: String,
    val title: String,
    val company: CompanyDetails,
    val personality: Personality,
    val skills: List<SkillInfoType>,
    val evaluations: List<Evaluation>,
    val teams: List<Team>
)

data class EmployeePerformance(
    val id: Int,
    val fullName: String,
    val result: Int
)

data class EvaluationEmployeeRequest(
    val date: String,
    val result: Int
)

data class EmployeesResponse(
    val success: Boolean,
    val data: List<Employee>?)

data class EmployeeResponse(
    val success: Boolean,
    val data: Employee
)

fun deserializeEmployeesResponse(response: JSONObject): EmployeesResponse {
    val gson = Gson()
    return gson.fromJson(response.toString(), EmployeesResponse::class.java)
}

fun deserializeEmployeesError(response: JSONObject): Exception {
    val error = response.optString("errors")
    return Exception(error)
}

fun serializeEvaluationEmployeeRequest(requestBody: EvaluationEmployeeRequest): JSONObject{
    val jsonObject = JSONObject()
    jsonObject.put("date", requestBody.date)
    jsonObject.put("result", requestBody.result)
    return jsonObject
}

fun deserializeEmployeeResponse(response: JSONObject): EmployeeResponse{
    val gson = Gson()
    return gson.fromJson(response.toString(), EmployeeResponse::class.java)
}

fun deserializeEmployeeError(error: JSONObject): Exception{
    val error = error.optString("errors")
    return Exception(error)
}